src/Microsoft.AspNetCore.OData/Query/Expressions/AggregationBinder.cs

```diff
@@ -8,235 +8,88 @@
-                            throw new NotImplementedException();
-                        }
-                    }
-
-                    _groupByClrType = typeof(GroupByWrapper);
-                    ResultClrType = typeof(AggregationWrapper);
-                    break;
-                default:
-                    throw new NotSupportedException(String.Format(CultureInfo.InvariantCulture,
-                        SRResources.NotSupportedTransformationKind, transformation.Kind));
+                throw Error.ArgumentNull(nameof(transformationNode));
             }
 
-            _groupByClrType = _groupByClrType ?? typeof(NoGroupByWrapper);
-        }
-
-        private static Expression WrapDynamicCastIfNeeded(Expression propertyAccessor)
-        {
-            if (propertyAccessor.Type == typeof(object))
+            if (context == null)
             {
-                return Expression.Call(null, ExpressionHelperMethods.ConvertToDecimal, propertyAccessor);
+                throw Error.ArgumentNull(nameof(context));
             }
 
-            return propertyAccessor;
-        }
-
-        private IEnumerable<AggregateExpressionBase> FixCustomMethodReturnTypes(IEnumerable<AggregateExpressionBase> aggregateExpressions)
-        {
-            return aggregateExpressions.Select(x =>
+            LambdaExpression groupLambda = null;
+            IEnumerable<GroupByPropertyNode> groupingProperties = GetGroupingProperties(transformationNode);
+            if (groupingProperties != null && groupingProperties.Any())
             {
-                var ae = x as AggregateExpression;
-                return ae != null ? FixCustomMethodReturnType(ae) : x;
-            });
-        }
+                // Generates expression
+                // .GroupBy($it => new DynamicTypeWrapper()
+                //                                      {
+                //                                           GroupByContainer => new AggregationPropertyContainer() {
+                //                                               Name = "Prop1",
+                //                                               Value = $it.Prop1,
+                //                                               Next = new AggregationPropertyContainer() {
+                //                                                   Name = "Prop2",
+                //                                                   Value = $it.Prop2, // int
+                //                                                   Next = new LastInChain() {
+                //                                                       Name = "Prop3",
+                //                                                       Value = $it.Prop3
+                //                                                   }
+                //                                               }
+                //                                           }
+                //                                      })
```

Comment:
How about you reduce the unnecessary whitespace so someone doesn't have to scroll horizontally ...
```suggestion
                // Generates expression
                // .GroupBy($it => new DynamicTypeWrapper()
                // {
                //     GroupByContainer => new AggregationPropertyContainer() {
                //         Name = "Prop1",
                //         Value = $it.Prop1,
                //         Next = new AggregationPropertyContainer() {
                //             Name = "Prop2",
                //             Value = $it.Prop2, // int
                //             Next = new LastInChain() {
                //                 Name = "Prop3",
                //                 Value = $it.Prop3
                //             }
                //         }
                //     }
                // })
```